settings.gradle

```diff
@@ -1,3 +1,19 @@
+pluginManagement {
```

Comment:
I have a hypothesis that use of the `plugins` block allows caching of plugin dependency resolution that isn't happening right now - `Resolving :classpath` on the first project of this build always takes a second, while on `-java` project it takes no time after the first run. Switching to `plugins` block will require inverting build logic, instead of `apply from: java.gradle -> apply plugin: 'java-library'`, it will be `plugins { java-library } -> plugins.withId('java-library) { apply from: 'java.gradle'`. I've generally found this pattern to have less evaluation order problems and prefer it anyways but it will be a big change so not trying yet.